diesel/src/pg/backend.rs

```diff
@@ -9,10 +9,41 @@ use prelude::Queryable;
 use query_builder::bind_collector::RawBytesBindCollector;
 use sql_types::{Oid, TypeMetadata};
 
+use std::slice;
+use std::ptr::NonNull;
+
 /// The PostgreSQL backend
 #[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
 pub struct Pg;
 
+#[derive(Debug)]
+pub struct PgValue {
```

Comment:
Is there any reason why we don't use a slice here?
Beside of that: Oid's should be `u32` (http://docs.diesel.rs/diesel/pg/types/sql_types/struct.Oid.html)
So that `PgValue would look like this:

```rust
pub struct PgValue<'a> {
    bytes: &'a [u8],
    oid: u32,
}

pub struct PgValueProxy;

impl<'a> FamilyLt<'a> for PgValueProxy {
    type Out = PgValue<'a>;
}
```

Reply:
Mostly because above we were discussing using NonNull :)